osh/worker/tasks/task_build.py

```diff
@@ -54,20 +55,28 @@ def run(self):
             self.spawn_subtask(*base_task_args, inherit_worker=True)
             self.wait()
 
+        # download custom mock config from the hub
+        mock_config_url = None
+        if mock_config == 'auto':
```

Comment:
do you think it necessary to update the command line helper to make it explicit? `mock-build`, `diff-build` etc. If I understand correctly it still requires to specify `--config=auto` from client otherwise falls to the default mock config which is populated from `DEFAULT_MOCKCONFIG` from client or system conf settings.

Moreover should we promote `DEFAULT_MOCKCONFIG = auto` from now on? Just out of curiosity.

Reply:
I'd rather keep this feature hidden for now because it is still quite experimental.

> Moreover should we promote DEFAULT_MOCKCONFIG = auto from now on? Just out of curiosity.

I don't think so, this feature needs actual Koji/Brew builds to exist so it is incompatible with SRPMs uploaded through the client.

Reply:
I believe that `--config=auto` will be the default one day.  But a few more steps need to be completed to get there.  We need to introduce some automagic rules to guess the correct profile for uploaded SRPMs and to use `cspodman` for Brew builds of containers (and ideally move the cspodman-specific bits to `osh-rh-conf`).  For now we need to stabilize the initial implementation before we proceed with the next steps.